ocaml/xcp-rrdd/bin/rrdd/xcp_rrdd.ml

```diff
@@ -347,6 +347,37 @@ let dss_loadavg () =
     )
   ]
 
+let dss_hostload xc domains =
+  let physinfo = Xenctrl.physinfo xc in
+  let pcpus = physinfo.Xenctrl.nr_cpus in
+  let rec loop n f = if n < 0 then 0 else f n + loop (n - 1) f in
```

Comment:
I would call this `sum` or `total`- the result is the sum of `f n` + `f (n-1)` + `f (n-2)` + ..; specifically, this is used to compute the number of active vcpus for every domain.

Reply:
This is not tail recursive - so `n` should not be larger than 100 or so